utils/oval/oval.go

```diff
@@ -0,0 +1,426 @@
+}
+
+// OvalFetcher implements updater.Fetcher.
+type OvalFetcher struct {
+	// OsInfo contains specifics to each Linux Distribution (see below)
+	OsInfo OSInfo
+}
+
+// OSInfo interface contains specifics methods for parsing OVAL definitions
+// that must be implemented by each Linux Distribution that uses OVAL
+// i.e. Red Hat and SUSE
+type OSInfo interface {
+	// ParsePackageNameVersion should, given a comment in a criterion, return
+	// the name and the version of the package.
+	// For example, if the comment is
+	//   glibc is earlier than 3.2
+	// should return glibc and 3.2
+	// This is based on the assumption that the distributions generate the comments
+	// automatically and they won't change (I know, not very reliable...)
+	ParsePackageNameVersion(comment string) (string, string)
+	// ParseOsVersion should, given a comment in a criterion, return the version
+	// of the Operating System.
+	// For example, if the comment is
+	//   SUSE Linux Enterpise Server 12 is installed
+	// should return 12
+	// This is based on the assumption that the distributions generate the comments
+	// automatically and they won't change it (I know, not very reliable...)
+	ParseOsVersion(comment string) string
+	// Given a line, parse for the xml file that contains the oval definition and
+	// returns the filename
+	// For example if the line contains
+	//	 com.redhat.rhsa-2003.xml, this will be return
+	// This is being used in conjunction with OvalUri (see below). Oval Uri contains
+	// a list of files, and you need ParseFilenameDist to get the right ones
+	ParseFilenameDist(line string) string
+	// OvalUri returns the url where the oval definitions are stored for a given
+	// distributions. See examples:
+	//   https://www.redhat.com/security/data/oval/
+	//   http://ftp.suse.com/pub/projects/security/oval/
+	OvalURI() string
+	// DistName returns the distribution name. Mostly used for debugging purposes
+	DistName() string
+	// IgnoredCriterions returns a list of strings that must be ignored when
+	// parsing the criterions.
+	// Oval parses parses all criterions by default trying to identify either
+	// package name and version or distribution version
+	IgnoredCriterions() []string
+	// SecToken returns a string that is compared with the value of reference.source
+	// in order to know if that is a security reference for, for example, using
+	// its url value
+	// Example return values: CVE, RHSA
+	SecToken() string
+	//Namespace stores the namespace that will be used in clair to store the vulnerabilities
+	Namespace() string
+}
+
+var (
+	log = capnslog.NewPackageLogger("github.com/coreos/clair", "utils/oval")
+)
+
+// FetchUpdate gets vulnerability updates from the OVAL definitions.
+func (f *OvalFetcher) FetchUpdate(datastore database.Datastore) (resp updater.FetcherResponse, err error) {
+	log.Info("fetching %s vulnerabilities", f.OsInfo.DistName())
+
+	if err != nil {
+		return resp, err
+	}
+
+	r, err := http.Get(f.OsInfo.OvalURI())
+
```

Comment:
remove this newline -- error checks should always be immediately following a call that returns an error